cfme/tests/infrastructure/test_rest_automation_request.py

```diff
@@ -64,7 +64,19 @@ def create_requests(collection, rest_api, automation_requests_data, multiple):
 
 
 def create_pending_requests(collection, rest_api, requests_pending):
+    # Wait to see that the `approval_state` does NOT change.
+    # It's enough to wait just for the first request, it gives
+    # other requests the same amount of time to change state.
+    waiting_request = requests_pending[0]
+    wait_for(
+        lambda: waiting_request.approval_state != 'pending_approval',
```

Comment:
how likely is the situation occurring where one of the other requests may not be complete?

i would believe its not that much of a pain to wait for the set of all approval states to hit it actually
`'pending_approval' not in set(request.approval_state for request in pending_requests)` as lambda expression seems simple at first glance, is there a finer detail making that one too expensive? such as reloading the requests